engine/core/src/streaming.c

```diff
@@ -91,6 +91,7 @@ uint16_t Stream_PutSample(streaming_channel_t *stream, const void *data, uint16_
 uint16_t Stream_GetSample(streaming_channel_t *stream, void *data, uint16_t size)
 {
     uint16_t nb_available_samples = Stream_GetAvailableSampleNB(stream);
+    LUOS_ASSERT(nb_available_samples >= size);
```

Comment:
If you put an assert here , the if statement is useless and the else case disappears

Reply:
You are right. This one is a mistake we should not make it...